sql/core/src/test/scala/org/apache/spark/sql/execution/python/PythonStreamingDataSourceSuite.scala

```diff
@@ -326,8 +326,11 @@ class PythonStreamingDataSourceSuite extends PythonDataSourceSuiteBase {
         lastBatch = q.lastProgress.batchId.toInt
       }
       assert(lastBatch > 20)
+      val rowCount = spark.read.format("json").load(outputDir.getAbsolutePath).count()
+      // There may be one uncommitted batch that is not recorded in query progress.
+      assert(rowCount == 2 * lastBatch + 2 || rowCount == 2 * lastBatch + 4)
```

Comment:
Can we explain the `2 * lastBatch + 2` and `2 * lastBatch + 4` in the comment here? Just curious why can't we provide an upper bound for the row count here?

Reply:
More comment added. We can check an upper bound here, but it is not really meaningful.